packages/dcos-checks/extra/dcos-checks/cmd/root.go

```diff
@@ -0,0 +1,214 @@
+		}
+
+		// init http client
+		var err error
+		httpClient, err = client.NewClient(nil, DCOSConfig.IAMConfig, DCOSConfig.CACert)
+		if err != nil {
+			logrus.Fatalf("Unable to initialize http client: %s", err)
+		}
+
+		if err := DCOSConfig.validateRequiredFlags(httpClient); err != nil {
+			logrus.Fatal(err)
+		}
+
+	},
+}
+
+// Execute adds all child commands to the root command sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	// run the commands parser
+	if err := RootCmd.Execute(); err != nil {
+		logrus.Fatalf("Error parsing subcommands: %s", err)
+	}
+}
+
+func init() {
+	cobra.OnInitialize(initConfig)
+	RootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/.checks.yaml)")
+
+	RootCmd.PersistentFlags().BoolVar(&DCOSConfig.ForceTLS, "force-tls", false, "use HTTPS for GET/POST requests")
+	RootCmd.PersistentFlags().BoolVar(&DCOSConfig.Verbose, "verbose", false, "enable verbose output")
+	RootCmd.PersistentFlags().StringVar(&DCOSConfig.Role, "role", "", "set DC/OS role. (valid roles: master, agent, public-agent)")
+	RootCmd.PersistentFlags().StringVar(&DCOSConfig.IAMConfig, "iam-config", "", "a path to identity and access managment config")
+	RootCmd.PersistentFlags().StringVar(&DCOSConfig.CACert, "ca-cert", "", "a path to certificate authority file")
+	RootCmd.PersistentFlags().StringVar(&DCOSConfig.DetectIP, "detect-ip", "/opt/mesosphere/bin/detect_ip", "a path to detect ip script")
+	RootCmd.PersistentFlags().StringVar(&DCOSConfig.NodeIPStr, "node-ip", "", "set node IP address overriding detect_ip output")
+}
+
+// initConfig reads in config file and ENV variables if set.
+func initConfig() {
+	if cfgFile != "" { // enable ability to specify config file via flag
+		viper.SetConfigFile(cfgFile)
+	}
+
+	viper.SetConfigName(".checks") // name of config file (without extension)
+	viper.AddConfigPath("$HOME")   // adding home directory as first search path
+	viper.AutomaticEnv()           // read in environment variables that match
+
+	// If a config file is found, read it in.
+	if err := viper.ReadInConfig(); err == nil {
+		fmt.Println("Using config file:", viper.ConfigFileUsed())
+	}
+}
+
+// DCOSChecker defines an interface for a generic DC/OS check.
+// ID() returns a check unique ID and RunCheck(...) returns a combined stdout/stderr, exit code and error.
+type DCOSChecker interface {
+	ID() string
+	Run(context.Context, *CLIConfigFlags) (string, int, error)
+}
+
+// RunCheck is a helper function that takes a list of DC/OS checks and runs checks one by one.
+func RunCheck(checks []DCOSChecker) {
+
+	exitCode := statusOK
+
+	for _, check := range checks {
+		output, retCode, err := check.Run(nil, DCOSConfig)
+		if err != nil {
+			logrus.Fatalf("Error executing %s: %s", check.ID(), err)
+		}
+		if retCode != statusOK {
+			exitCode = statusFailure
+		}
+		if output != "" {
+			fmt.Printf("[%s]: %s\n", check.ID(), output)
```

Comment:
this is intensional, the output would be a custom format, i need to place TODO here